src/monitord/monitor_actions.c

```diff
@@ -0,0 +1,256 @@
+/* Copyright (C) 2015-2020, Wazuh Inc.
+ * Copyright (C) 2009 Trend Micro Inc.
+ * All right reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation
+ */
+
+#include "shared.h"
+#include "monitord.h"
+#include "read-agents.h"
+#include "wazuh_db/wdb.h"
+
+static int mon_send_agent_msg(char *agent, char *msg);
+int sock = -1;
+
+void monitor_send_deletion_msg(char *agent) {
+    char str[OS_SIZE_1024 + 1];
+
+    memset(str, '\0', OS_SIZE_1024 + 1);
```

Comment:
Not sure whether is necessary the `+1` and `-1`. According to the [snprintf documentation](http://www.cplusplus.com/reference/cstdio/snprintf/), the generated string is `n-1` and the remaining character is the null character that determines the end of the string. So, I think that doing this should be enough...

```c
char str[OS_SIZE_1024];
memset(str, '\0', OS_SIZE_1024);
snprintf(str, OS_SIZE_1024, OS_AG_REMOVED, agent);
```

Reply:
You are right. Updated!